flint/flint-spark-integration/src/main/scala/org/opensearch/flint/spark/FlintSpark.scala

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.flint.spark
+
+import scala.collection.JavaConverters._
+
+import org.opensearch.flint.core.{FlintClient, FlintOptions}
+import org.opensearch.flint.core.FlintOptions._
+import org.opensearch.flint.core.metadata.FlintMetadata
+import org.opensearch.flint.core.storage.FlintOpenSearchClient
+import org.opensearch.flint.spark.FlintSpark._
+import org.opensearch.flint.spark.skipping.{FlintSparkSkippingIndex, FlintSparkSkippingStrategy}
+import org.opensearch.flint.spark.skipping.partition.PartitionSkippingStrategy
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalog.Column
+
+/**
+ * Flint Spark integration API entrypoint.
+ */
+class FlintSpark(val spark: SparkSession) {
+
+  /** Flint client for low-level index operation */
+  private val flintClient: FlintClient = {
+    val options = new FlintOptions(
+      Map(
+        HOST -> spark.conf.get(FLINT_INDEX_STORE_LOCATION, FLINT_INDEX_STORE_LOCATION_DEFAULT),
+        PORT -> spark.conf.get(FLINT_INDEX_STORE_PORT, FLINT_INDEX_STORE_PORT_DEFAULT)).asJava)
+    new FlintOpenSearchClient(options)
+  }
+
+  /**
+   * Create index builder for creating index with fluent API.
+   *
+   * @return
+   *   index builder
+   */
+  def skippingIndex(): IndexBuilder = {
+    new IndexBuilder(this)
+  }
+
+  /**
+   * Create the given index with metadata.
+   *
+   * @param index
+   *   Flint index to create
+   */
+  def createIndex(index: FlintSparkIndex): Unit = {
+    val indexName = index.name()
+    if (flintClient.exists(indexName)) {
+      throw new IllegalStateException(
+        s"A table can only have one Flint skipping index: Flint index $indexName is found")
+    }
+    flintClient.createIndex(indexName, index.metadata())
+
+    // TODO: start building index by Flint data source write capability
+  }
+
+  /**
+   * Describe a Flint index.
+   *
+   * @param indexName
+   *   index name
+   * @return
+   *   Flint index metadata
+   */
+  def describeIndex(indexName: String): Option[FlintMetadata] = {
+    if (flintClient.exists(indexName)) {
+      Some(flintClient.getIndexMetadata(indexName))
+    } else {
+      Option.empty
+    }
+  }
+
+  /**
+   * Delete index.
+   *
+   * @param indexName
+   *   index name
+   * @return
+   *   true if exist and deleted, otherwise false
+   */
+  def deleteIndex(indexName: String): Boolean = {
+    if (flintClient.exists(indexName)) {
+      flintClient.deleteIndex(indexName)
+      true
+    } else {
+      false
+    }
+  }
+}
+
+object FlintSpark {
+
+  /**
+   * Flint configurations in Spark. TODO: shared with Flint data source config?
+   */
+  val FLINT_INDEX_STORE_LOCATION = "spark.flint.indexstore.location"
```

Comment:
QQ, should we put all the flint spark related configuration in here?

Reply:
yes, I think we should put all together here or a new `FlintSparkConf` class later.